app/src/main/java/com/hedvig/app/feature/offer/ui/OfferActivity.kt

```diff
@@ -152,28 +156,45 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
                                 setSignState(viewState.signMethod)
                             }
                             is OfferViewModel.ViewState.Error.GeneralError -> showErrorDialog(
-                                viewState.message ?: getString(R.string.home_tab_error_body)
+                                viewState.message ?: getString(R.string.home_tab_error_body),
+                                ::finish
                             )
                             is OfferViewModel.ViewState.Error -> showErrorDialog(
-                                getString(R.string.home_tab_error_body)
+                                getString(R.string.home_tab_error_body),
+                                ::finish
                             )
                             is OfferViewModel.ViewState.Loading -> topOfferAdapter.submitList(viewState.loadingItem)
                         }
                     }
             }
-
-            signButton.setHapticClickListener {
-                tracker.floatingSign()
-                OfferSignDialog.newInstance().show(
-                    supportFragmentManager,
-                    OfferSignDialog.TAG
-                )
-            }
         }
     }
 
     private fun setSignState(signMethod: SignMethod) {
-        // Set sign state
+        binding.signButton.bindWithSignMethod(signMethod)
+        onSign(signMethod)
+    }
+
+    private fun onSign(signMethod: SignMethod) {
+        when (signMethod) {
+            SignMethod.SWEDISH_BANK_ID -> {
+                binding.signButton.setHapticClickListener {
+                    tracker.floatingSign()
+                    OfferSignDialog.newInstance().show(
+                        supportFragmentManager,
+                        OfferSignDialog.TAG
+                    )
+                }
```

Comment:
This does not seem right I think - shouldn't we be triggering the `OfferSignDialog` in this case instead of setting up the click listener? 😬

Reply:
Yes 😅 This is fixed in APP-543!